pkg/services/rendering/plugin_mode.go

```diff
@@ -35,7 +44,46 @@ func (rs *RenderingService) renderViaPlugin(ctx context.Context, renderKey strin
 	}
 	rs.log.Debug("calling renderer plugin", "req", req)
 
-	rsp, err := rs.pluginInfo.GrpcPlugin.Render(ctx, req)
+	rsp, err := rs.pluginInfo.GrpcPluginV1.Render(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	if rsp.Error != "" {
+		return nil, fmt.Errorf("Rendering failed: %v", rsp.Error)
+	}
+
+	return &RenderResult{FilePath: pngPath}, err
```

Comment:
Better to be explicit about `err` being nil.
```suggestion
	return &RenderResult{FilePath: pngPath}, nil
```